src/common/Database/DatabaseWorkerPool.cpp

```diff
@@ -99,6 +95,82 @@ void DatabaseWorkerPool<T>::Close()
     sLog->outSQLDriver("All connections on DatabasePool '%s' closed.", GetDatabaseName());
 }
 
+template <class T>
+uint32 DatabaseWorkerPool<T>::OpenConnections(InternalIndex type, uint8 numConnections)
+{
+    _connections[type].resize(numConnections);
+    for (uint8 i = 0; i < numConnections; ++i)
+    {
+        T* t;
+
+        if (type == IDX_ASYNC)
+            t = new T(_queue, *_connectionInfo);
+        else if (type == IDX_SYNCH)
+            t = new T(*_connectionInfo);
+        else
+            ABORT();
+
+        _connections[type][i] = t;
+        ++_connectionCount[type];
+
+        uint32 error = t->Open();
+
+        if (!error)
+        {
+            if (mysql_get_server_version(t->GetHandle()) < MIN_MYSQL_SERVER_VERSION)
```

Comment:
please make sure it works with all versions of MariaDB, including the ones based on MySQL 5.5

Reply:
Mysql has dropped support for 5.5 and 5.6, and so have we long ago.

Reply:
Work with MariaDB 10.3 - 10.5